packages/itwinui-react/src/core/Modal/Modal.test.tsx

```diff
@@ -240,3 +240,25 @@ it('should reset body overflow on closing and unmounting', () => {
   unmount();
   expect(document.body.style.overflow).not.toEqual('hidden');
 });
+
+it('should allow passing arbitrary props to internal elements', () => {
+  render(
+    <Modal
+      isOpen={true}
+      title='The title'
+      className='modal'
+      wrapperProps={{ className: 'wrapper' }}
+      backdropProps={{ className: 'backdrop' }}
+      titleBarProps={{ className: 'title-bar' }}
+    >
+      …
+    </Modal>,
+  );
+
+  const wrapper = document.querySelector('.wrapper') as HTMLElement;
+  const backdrop = wrapper.querySelector('.backdrop') as HTMLElement;
+  expect(backdrop).toBeTruthy();
+  const modal = wrapper.querySelector('.modal') as HTMLElement;
+  const titleBar = modal.querySelector('.title-bar') as HTMLElement;
```

Comment:
nit: Is there any advantage of spliting into `modal` and `titleBar` instead of directly using `titleBar`?

```
const titleBar = wrapper.querySelector('.modal > .title-bar') as HTMLElement;
```

Reply:
Sounds good, I was just curious. Then how it is is good.